gtsam/hybrid/HybridConditional.h

```diff
@@ -172,7 +174,8 @@ class GTSAM_EXPORT HybridConditional
    */
   template <typename T = DiscreteConditional>
   typename T::shared_ptr asDiscrete() const {
-    return std::dynamic_pointer_cast<T>(inner_);
+    if (!isDiscrete()) return nullptr;
```

Comment:
Preferably throw exception instead of returning a nullptr since this can be a silent error/failure.

Reply:
No, @varunagrawal , this is used to find out whether it is of this type. I don’t think we should mess with that semantics